src/models/legend.js

```diff
@@ -176,8 +177,22 @@ nv.models.legend = function() {
             if (align) {
 
                 var seriesWidths = [];
+                var largestKey = 0;
+
+                series.each(function(d,i) {
+                  if (getKey(d).length > largestKey) {
+                    largestKey = getKey(d).length;
+                  }
+                });
+
                 series.each(function(d,i) {
-                    var legendText = d3.select(this).select('text');
+                    var legendText;
+                    if (largestKey > maxKeyLength && getKey(d).length > maxKeyLength) { 
```

Comment:
Correct, initially largest text was only taking into account one key; however, getKey(d).length will account for all the keys in the legend. I agree, I can remove largestKey as its redundant. Will resubmit a patch.